### PR TITLE
Extend formatter script to run on all staged files, merge add/strip action

### PR DESCRIPTION
**Pull request by tcm-marcel**
This is especially useful when using the new pre-commit hook that prevents me from committing. 

* add -f option to formatter that applies the selected action to all staged files (git)
* adapt pre-commit hook to only check stages files

(since -a, -s and -c were already taken, I just picked any other letter that somehow looks like "staged files" when looking from far away)

*Testing:* I run Ubuntu, can someone test this on a Mac?

### CODE REVIEW

**Comment by tcm-marcel** [comment]
There was a type in the commit message and in the title - of course it is meant to run on all *staged* files, not the other way round.

**Comment by schedutron** [comment]
Perhaps a lame question, but *how exactly* do I test this? I'm on a Mac.

**Comment by tcm-marcel** [comment]
@schedutron 
1. Install the [pre-commit hook](https://github.com/cmu-db/peloton/tree/master/script/git-hooks)
2. Edit some source or header files in peloton to obviously violate the code style (e.g. lines too long)
3. Add some (not all) of them to the staging area (`git add`) and try to commit. The pre-commit hook should prevent this. 
4. run `./script/formatting/formatter.py -c -f` - the staged files should be formatted now and ready to commit, the unstaged ones should remain unchanged

Thanks!

**Comment by schedutron** [comment]
I installed the hook and edited few source files (made some of the lines longer that 80 chars), staged them and committed some of them, but the hook didn't stop me. Rather, I got this commit output:
```
[pr/formatter 8c9d66ad4] Check formatter
 1 file changed, 1 insertion(+), 3 deletions(-)
```

**Comment by tcm-marcel** [comment]
Tanks for testing!

> And, is there a way to only run the hook if there are staged files? I tried to commit without staging and the hook still ran.
The commit hook only checks staged files. But there is no way to temporary disable it, if that's what you mean.

**Comment by schedutron** [comment]
@tcm-marcel Maybe the formatter can be edited to only run when there are nonzero files passed to the `--files` parameter?

**Comment by tcm-marcel** [comment]
@schedutron was right, I changed the hook to only run if actually .cpp/.h files are staged. If committing a non-source file, the validator would run with empty parameters otherwise. 

Referring to last team meeting I also merged the add/strip actions to a new update action, that updates the header or adds a new one if necessary.

File: script/git-hooks/pre-commit [hunk 1]
```diff
@@ -8,6 +8,11 @@
 # To enable, symlink this file to '.git/hooks/pre-commit' like so:
 #   ln -s ../../script/git-hooks/pre-commit .git/hooks/pre-commit
 
-files=$(git diff --name-only HEAD --diff-filter=d | grep '\.\(cpp\|h\)$')
-python script/validators/source_validator.py --files $files
-exit $?
+files=$(git diff --name-only HEAD --cached --diff-filter=d | grep '\.\(cpp\|h\)$')
+echo $files
```
Comment (pmenon) [inline script/git-hooks/pre-commit]: You can remove this `echo` since all scanned files are printed in the `source_validator.py` script.

**Comment by schedutron** [comment]
@pmenon So you're suggesting that @tcm-marcel should add a `try-except` block to check for the `clang-format` command, right?

File: script/git-hooks/pre-commit (hunk 1 above)
Comment (tcm-marcel) [reply script/git-hooks/pre-commit]: oops, that one is left from debugging

**Pull request merged**